api/v1alpha1/nginxingresscontroller_types.go

```diff
@@ -50,6 +50,28 @@ type NginxIngressControllerSpec struct {
 	// will be from the Azure LoadBalancer annotations here https://cloud-provider-azure.sigs.k8s.io/topics/loadbalancer/#loadbalancer-annotations
 	// +optional
 	LoadBalancerAnnotations map[string]string `json:"loadBalancerAnnotations,omitempty"`
+
+	// DefaultSSLCertificate is a struct with a secret with the fields namespace and name which is used to create the ssl certificate used by the default HTTPS server
+	// +optional
+	DefaultSSLCertificate *DefaultSSLCertificate `json:"defaultSSLCertificate,omitempty"`
+}
+
+type DefaultSSLCertificate struct {
+	// Secret is a struct that holds the name and namespace fields used for the default ssl secret
+	// +optional
+	Secret *Secret `json:"secret"`
```

Comment:
can we use CEL to validate that one of the fields in the DefaultSSLCertificate is filled out?

https://opensource.googleblog.com/2023/11/kubernetes-crd-validation-using-cel.html

kubebuilder:validation:XValidation:rule is the comment